src/CommonLib/LDAPUtils.cs

```diff
@@ -979,99 +1018,15 @@ private LdapConnection CreateNewConnection(string domainName = null, bool global
-                {
-                    retryCount++;
-                    Thread.Sleep(backoffDelay);
-                    backoffDelay = TimeSpan.FromSeconds(Math.Min(
-                        backoffDelay.TotalSeconds * BackoffDelayMultiplier.TotalSeconds, MaxBackoffDelay.TotalSeconds));
-                    conn = CreateNewConnection(domainName, globalCatalog, skipCache);
-                    if (conn == null)
-                    {
-                        _log.LogError("Unable to create replacement ldap connection for ServerDown exception. Breaking loop");
-                        yield break;
-                    }
-                    
-                    _log.LogInformation("Created new LDAP connection after receiving ServerDown from server");
-                    continue;
-                }
-                catch (LdapException le)
-                {
-                    if (le.ErrorCode != 82)
-                        if (throwException)
-                            throw new LDAPQueryException(
-                                $"LDAP Exception in Loop: {le.ErrorCode}. {le.ServerErrorMessage}. {le.Message}. Filter: {ldapFilter}. Domain: {domainName}",
-                                le);
-                        else
-                            _log.LogWarning(le,
-                                "LDAP Exception in Loop: {ErrorCode}. {ServerErrorMessage}. {Message}. Filter: {Filter}. Domain: {Domain}",
-                                le.ErrorCode, le.ServerErrorMessage, le.Message, ldapFilter, domainName);
-                    yield break;
-                }
-                catch (Exception e)
-                {
-                    if (throwException)
-                        throw new LDAPQueryException(
-                            $"Exception in LDAP loop for {ldapFilter} and {domainName ?? "Default Domain"}", e);
-
-                    _log.LogWarning(e, "Exception in LDAP loop for {Filter} and {Domain}", ldapFilter,
-                        domainName ?? "Default Domain");
-                    yield break;
-                }
-
-                if (response == null || pageResponse == null) continue;
-
-                if (response.Entries == null)
-                    yield break;
-
-                foreach (SearchResultEntry entry in response.Entries)
-                    yield return new SearchResultEntryWrapper(entry, this);
-
-                if (pageResponse.Cookie.Length == 0 || response.Entries.Count == 0)
-                    yield break;
+            return QueryLDAP(ldapFilter, scope, props, new CancellationToken(), domainName, includeAcl, showDeleted,
+                adsPath, globalCatalog, skipCache, throwException);
+        }
 
-                pageControl.Cookie = pageResponse.Cookie;
-            }
+        private static TimeSpan GetNextBackoff(int retryCount)
+        {
+            return TimeSpan.FromSeconds(Math.Min(
+                BackoffDelayMultiplier * (retryCount + 1) * retryCount,
```

Comment:
This formula is fine but usually you'll see backoff calculation as something like this:

``` c#
TimeSpan.FromSeconds(Math.Min(
  MinBackoffDelay.TotalSeconds * Math.Pow(BackoffDelayMultiplier, retryCount),
  MaxBackoffDelay.TotalSeconds));
```

I believe they call this a truncated exponential backoff. You might consider adopting it and gate retrying the connection exclusively on the maximum number of retry attempts. That way you can simplify gating your retry attempts on just one variable and conform to a standard way of doing things.